lib/core/versions/latest/CreateOperation.ts

```diff
@@ -150,6 +152,27 @@ export default class CreateOperation implements OperationModel {
     Multihash.verifyHashComputedUsingLatestSupportedAlgorithm(deltaHash);
     Multihash.verifyHashComputedUsingLatestSupportedAlgorithm(nextRecoveryCommitment);
 
+    // type has to be max 4 character long string with only base64url character set
+    if (properties.length === 3) {
+      if (typeof suffixData.type !== 'string') {
+        throw new SidetreeError(ErrorCode.CreateOperationSuffixDataTypeIsNotString);
+      }
+
+      if (suffixData.type.length > 4) {
+        throw new SidetreeError(ErrorCode.CreateOperationSuffixDataTypeLengtGreaterThanFour);
+      }
+
+      if (!/^[A-Za-z0-9_-]+$/.test(suffixData.type)) {
+        throw new SidetreeError(ErrorCode.CreateOperationSuffixDataTypeInvalidCharacter);
+      }
+
+      return {
+        deltaHash: suffixData.delta_hash,
+        recoveryCommitment: suffixData.recovery_commitment,
+        type: suffixData.type
+      };
```

Comment:
Nit: I believe technically you don't need to dup the return statement here, adding `type: suffixData.type` in the existing return should be sufficient.